docs/cow-protocol/tutorials/cow-swap/flash-loans.mdx

```diff
@@ -0,0 +1,119 @@
+- Tokens in Aave can be [obtained from the faucet](https://github.com/mrnaveira/cow-flashloan-tester?tab=readme-ov-file#5-get-tokens-from-the-faucet).
+- Check [withdrawal and borrowing availability](https://github.com/mrnaveira/cow-flashloan-tester?tab=readme-ov-file#7-check-aaves-withdrawalborrow-availability) of the tokens on Aave.
+- Try a different RPC in case of transaction submission issues.
+
+In a production network (e.g. mainnet), these considerations generally won’t apply.
+
+:::
+
+### Order construction
+
+```json5
+{
+  "from": "Safe address",
+  "sellToken": "USDC address", // collateral token (unlocked by repaying the debt)
+  "buyToken": "USDT address", // originally borrowed token that was now advanced by the flash loan
+  "buyAmount": 5002500000, // appData.flashloan.amount + 0.05% flashloan fee
+  "receiver": "settlementContract", // this is for repaying the flash loan
+  "validTo": "now + 5m", // managing risk can be done by having a short validity
+  "kind": "buy", // buy exactly the flash loaned amount and keep the surplus in the collateral token
+  "partiallyFillable": false, // if an order is partially fillable, then it is not ensured the debt will be paid
+  "appData": {
+    "hooks": {
+      "pre": [
+         // First: Repay the debt
+         {
+            "target": "0x123...321",
+            "value": "0",
+            "callData": "signedRepayCall"
+         },
+         // Then: Withdraw the collateral
+         {
+            "target": "0x123...321",
+            "value": "0",
+            "callData": "signedWithdrawCall"
+         }
+      ]
+    },
+    "flashloan": {
+      "token": "USDT",
+      "amount": 5000000000 // 5,000 USDT in atoms
+    }
+  }
+}
+```
+
+Check the [flash loan tester script](https://github.com/mrnaveira/cow-flashloan-tester) for more details on how this order is constructed.
+
+### Setting the pre-signature of the order
+
+After the order is placed, its pre-signature must be submitted to the CoW Settlement Contract via a transaction:
+* Function [`setPresignature`](https://docs.cow.fi/cow-protocol/reference/contracts/core/settlement#setpresignature).
+* Arguments: `orderId` and `true`.
+
+The [flash loan tester script](https://github.com/mrnaveira/cow-flashloan-tester) already sends this transaction automatically to be executed by the Safe. You can see an example of this transaction [here](https://sepolia.etherscan.io/tx/0x2de6e4cfcf61942f73e2db254a6b917fc52ec95da4d296f5a998be4c1b910c94).
+
+After the pre-signature transaction is confirmed, the order's status will be `Open`, and it will be eligible for execution within the CoW Protocol.
+
+### Order execution
+
+Once an order is placed within the CoW Protocol, it enters an auction batch. When a solution is found, the following steps occur:
+
+1. The winning solver calls the flash loan `IFlashLoanRouter` contract.
```

Comment:
I wonder if we could make this a bit more visual using a graphic like [here](https://app.blocksec.com/explorer/tx/sepolia/0x753b65e09447f1fb783b5aabbb4c1e35cf4311fb08a1d9695130e587a4f6812d).
We could color and number the individual token movements and explain what's happening at each step.

Reply:
IMO it's better to leave it for a follow-up PR